File: mcp/smoke-test.py
from __future__ import annotations

import asyncio
import json
from pathlib import Path

from mcp import ClientSession, StdioServerParameters
from mcp.client.stdio import stdio_client
from mcp.types import TextContent

PROJECT_ROOT = Path(__file__).resolve().parents[1]


def content_text(result: object) -> str:
    content = getattr(result, "content", [])
    for item in content:
        if isinstance(item, TextContent):
            return item.text
    return ""


def resource_text(result: object) -> str:
    contents = getattr(result, "contents", [])
    for item in contents:
        text = getattr(item, "text", None)
        if isinstance(text, str):
            return text
    return ""


async def main() -> None:
    server = StdioServerParameters(
        command="uv",
        args=["run", "--with", "mcp", "python", "mcp/server.py"],
        cwd=PROJECT_ROOT,
    )

    async with stdio_client(server) as (read_stream, write_stream):
        async with ClientSession(read_stream, write_stream) as session:
            await session.initialize()

            tools = await session.list_tools()
            print(f"tools: {', '.join(tool.name for tool in tools.tools)}")

            components = await session.call_tool(
                "list_components",
                {"query": "button"},
            )
            print(f"list_components sample:\n{content_text(components)[:1200]}")

            button = await session.call_tool(
                "get_component",
                {
                    "name": "ui/button",
                    "include_source": True,
                    "include_story": True,
                },
            )
            print(f"get_component sample:\n{content_text(button)[:1200]}")

            resource = await session.read_resource("teamhelper-ui-python://design-tokens")
            sample = resource_text(resource)
            parsed = json.loads(sample)
            print(
                "design-tokens resource sample:\n"
                f"{json.dumps(parsed, ensure_ascii=False, indent=2)[:1200]}"
            )


if __name__ == "__main__":
    asyncio.run(main())

File: mcp/smoke-test.ts
import { Client } from '@modelcontextprotocol/sdk/client/index.js';
import { StdioClientTransport } from '@modelcontextprotocol/sdk/client/stdio.js';
import { dirname, resolve } from 'node:path';
import { fileURLToPath } from 'node:url';

type TextToolResult = {
  content?: Array<{
    type?: string;
    text?: string;
  }>;
};

type TextResourceResult = {
  contents?: Array<{
    text?: string;
  }>;
};

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);
const projectRoot = resolve(__dirname, '..');

function hasContent(value: unknown): value is TextToolResult {
  return typeof value === 'object' && value !== null && 'content' in value;
}

function hasContents(value: unknown): value is TextResourceResult {
  return typeof value === 'object' && value !== null && 'contents' in value;
}

function firstToolText(result: unknown) {
  if (!hasContent(result)) {
    return '';
  }

  return result.content?.find((item) => item.type === 'text')?.text ?? '';
}

function firstResourceText(result: unknown) {
  if (!hasContents(result)) {
    return '';
  }

  return (
    result.contents?.find((item) => typeof item.text === 'string')?.text ?? ''
  );
}

async function main() {
  const transport = new StdioClientTransport({
    command: 'pnpm',
    args: ['exec', 'tsx', 'mcp/server.ts'],
    cwd: projectRoot,
    stderr: 'pipe',
  });

  transport.stderr?.on('data', (chunk) => {
    process.stderr.write(`[server] ${chunk}`);
  });

  const client = new Client({
    name: 'teamhelper-ui-mcp-smoke-test',
    version: '0.1.0',
  });

  try {
    await client.connect(transport);

    const tools = await client.listTools();
    console.log(`tools: ${tools.tools.map((tool) => tool.name).join(', ')}`);

    const components = await client.callTool({
      name: 'list_components',
      arguments: { query: 'button' },
    });
    console.log(
      `list_components sample:\n${firstToolText(components).slice(0, 1200)}`
    );

    const button = await client.callTool({
      name: 'get_component',
      arguments: {
        name: 'ui/button',
        includeSource: true,
        includeStory: true,
      },
    });
    console.log(
      `get_component sample:\n${firstToolText(button).slice(0, 1200)}`
    );

    const resource = await client.readResource({
      uri: 'teamhelper-ui://design-tokens',
    });
    console.log(
      `design-tokens resource sample:\n${firstResourceText(resource).slice(0, 1200)}`
    );
  } finally {
    await client.close();
  }
}

main().catch((error: unknown) => {
  console.error(error);
  process.exit(1);
});
